kie-remote/kie-remote-client/src/main/java/org/kie/services/client/api/command/AbstractRemoteCommandObject.java

```diff
@@ -228,15 +228,19 @@ void preprocessParameterCommand( Object cmdObj, List<Object> extraClassInstanceL
 
     void addPossiblyNullObject( Object inputObject, List<Object> objectList ) {
         if( inputObject != null ) {
-            if( inputObject instanceof List )  { 
-                objectList.addAll((List) inputObject);
+            if( inputObject instanceof List )  {
+                for( Object obj : (List<?>)inputObject ) {
+                    addPossiblyNullObject(obj, objectList);
+                }
```

Comment:
I have an.. algorithm to do this, so to speak. I'll merge it and add it to the code -- unless you want to fix it yourself? :)